### PR TITLE
Add button for displaying STR variants occurring in the dynamic HPO panel on case page

### LINKED ISSUE #4473: Clinical HPO STRs-button
Lund wishes for a "Clinical HPO STRs-button" similar to "Clinical HPO SNVs" and "Clinical HPO SVs", should be a minor thing?

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4473 

Main branch:

image

This branch:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
> Great! You will also be needing an `update_form_hgnc_symbols()` in `variants.controllers.str_variants()`. 😊

Yeah I saw that it's more complicated than this, I'll fix it tomorrow, thanks! 😅

**Comment by northwestwitch** [comment]
@dnil you were right, dynamic_hpo_panel_strs_link make the trick! Deployed on stage and tested with [this case](https://scout-stage.scilifelab.se/cust000/F0006358-mip7)

File: scout/server/blueprints/cases/templates/cases/gene_panel.html [hunk 1]
```diff
@@ -152,40 +152,58 @@ <h6 class="ms-3 mt-3"><span class="fas fa-clipboard-list"></span>&nbsp;HPO gene
...
-                               gene_panels=['hpo']) }}">
-               Clinical HPO SNVs
-             </a>
-            <a class="btn btn-secondary btn-sm text-white"
-              href="{{ url_for('variants.cancer_sv_variants', institute_id=institute._id,
-                               case_name=case.display_name, variant_type='clinical',
-                               gene_panels=['hpo']) }}">
-               Clinical HPO SVs
-             </a>
+            {% if case.vcf_files.vcf_cancer %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/gene_panel.html:158]: 👍

File: scout/server/blueprints/cases/templates/cases/gene_panel.html [hunk 2]
```diff
@@ -152,40 +152,58 @@ <h6 class="ms-3 mt-3"><span class="fas fa-clipboard-list"></span>&nbsp;HPO gene
...
-              href="{{ url_for('variants.sv_variants', institute_id=institute._id,
+            {% if case.vcf_files.vcf_snv %}
+              <a class="btn btn-secondary btn-sm text-white"
+                href="{{ url_for('variants.variants', institute_id=institute._id,
+                                 case_name=case.display_name, variant_type='clinical',
+                                 gene_panels=['hpo']) }}">
+                 Clinical HPO SNVs
+              </a>
+            {% endif %}
+            {% if case.vcf_files.vcf_sv %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/gene_panel.html:183]: Great to check: we do have an (incomplete?) set of bools on case like `case["has_strvariants"]` `case["has_svvariants"]` that are supposed to encapsulate that a little. 😊

File: scout/server/blueprints/variants/views.py [hunk 3]
```diff
@@ -195,6 +201,7 @@ def str_variants(institute_id, case_name):
         form.chrom.data = request.args.get("chrom", "")
 
     controllers.populate_force_show_unaffected_vars(institute_obj, form)
+    controllers.update_form_hgnc_symbols(store, case_obj, form)
```
Comment (dnil) [inline scout/server/blueprints/variants/views.py:204]: ✅

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 ## [unreleased]
 ### Added
 - Added tags for Sniffles and CNVpytor, two LRS SV callers
+- Button for displaying STR variants occurring in the dynamic HPO panel on case page
```
Comment (dnil) [inline CHANGELOG.md:10]:
```suggestion
- Button on case page for displaying STR variants occurring in the dynamic HPO panel
```

**Review by dnil — APPROVED** [review]
Works just fine! Code-wise I would suggest replacing the jinja case file var existence checks and fix the `has_` bool generation if I needed one or two more.

**Comment by northwestwitch** [comment]
> Works just fine! Code-wise I would suggest replacing the jinja case file var existence checks and fix the `has_` bool generation if I needed one or two more.

Good suggestion, I'll fix accordingly. I was also thinking that it would be nice if the links open in a new tab

**Pull request merged**